pkg/driverbuilder/builder/image.go

```diff
@@ -45,78 +63,123 @@ func (im ImagesMap) findImage(target Type, gccVers semver.Version) (Image, bool)
-	regs = append(regs, regexp.MustCompile(targetFmt))
-	genericFmt := fmt.Sprintf("driverkit-builder-any-%s(?P<gccVers>(_gcc[0-9]+.[0-9]+.[0-9]+)+)$", arch)
-	regs = append(regs, regexp.MustCompile(genericFmt))
-
-	b.Images = make(ImagesMap)
-	for _, repo := range b.BuilderRepos {
-		imgs, err := cli.ImageSearch(context.Background(), repo, types.ImageSearchOptions{Limit: 100})
-		if err != nil {
-			logger.Warnf("Skipping repo %s: %s\n", repo, err.Error())
-			continue
-		}
-		for _, img := range imgs {
-			for regIdx, reg := range regs {
-				match := reg.FindStringSubmatch(img.Name)
-				if len(match) == 0 {
-					continue
-				}
-
-				var gccVers []string
-				for i, name := range reg.SubexpNames() {
-					if i > 0 && i <= len(match) {
-						switch name {
-						case "gccVers":
-							gccVers = strings.Split(match[i], "_gcc")
-							gccVers = gccVers[1:] // remove initial whitespace
-						}
+			var gccVers []string
+			for i, name := range reg.SubexpNames() {
+				if i > 0 && i <= len(match) {
+					switch name {
+					case "gccVers":
+						gccVers = strings.Split(match[i], "_gcc")
+						gccVers = gccVers[1:] // remove initial whitespace
 					}
 				}
+			}
 
-				if len(gccVers) == 0 {
-					logger.Debug("Malformed image name: ", img.Name, len(match))
-					continue
-				}
+			if len(gccVers) == 0 {
+				logger.Debug("Malformed image name: ", img.Name, len(match))
+				continue
+			}
 
-				// Note: we store "any" target images as "any",
-				// instead of adding them to the target,
-				// because we always prefer specific target images,
-				// and we cannot guarantee here that any subsequent docker repos
-				// does not provide a target-specific image that offers same gcc version
-				for _, gccVer := range gccVers {
-					// If user set a fixed gcc version, only load images that provide it.
-					if b.GCCVersion != "" && b.GCCVersion != gccVer {
-						continue
-					}
-					buildImage := Image{
-						GCCVersion: mustParseTolerant(gccVer),
-						Name:       img.Name,
-					}
-					if regIdx == 0 {
-						buildImage.Target = b.TargetType
-					} else {
-						buildImage.Target = Type("any")
-					}
-					// Skip if key already exists: we have a descending prio list of docker repos!
-					if _, ok := b.Images[buildImage.toKey()]; !ok {
-						b.Images[buildImage.toKey()] = buildImage
-					}
+			// Note: we store "any" target images as "any",
+			// instead of adding them to the target,
+			// because we always prefer specific target images,
+			// and we cannot guarantee here that any subsequent docker repos
+			// does not provide a target-specific image that offers same gcc version
+			for _, gccVer := range gccVers {
+				// If user set a fixed gcc version, only load images that provide it.
+				buildImage := Image{
+					GCCVersion: mustParseTolerant(gccVer),
+					Name:       img.Name,
+				}
+				if regIdx == 0 {
+					buildImage.Target = Type("target-placeholder")
```

Comment:
I think the best here would be to properly capture the type in the regexes, so that we can just do `buildImage.Target = Type(regType)`
be it `any` or a specific target.

Reply:
And we can then skip the `target-placeholder` trick below.